File: tests/dat/things/TestChest.js
TestChest.doInit = function () {
  this.opened = false;
};

TestChest.onUse = function (user) {
  if (this.opened) {
    user.sendMessage("TestChest is already open.\n");
    return;
  }

  this.opened = true;
  user.sendMessage("You open the TestChest.\n");
  user.gainItem("TestToken");
};

TestChest.onInspect = function (inspector) {
  return "A simple chest used for tests.\n";
};

File: tests/ext/things/TestChest.lua
function TestChest:doInit()
    self.opened = false
end

function TestChest:onUse(event)
    if self.opened then
        event.actor:sendMessage("TestChest is already open.\n")
        return
    end

    self.opened = true
    event.actor:sendMessage("You open the TestChest.\n")
    event.actor:gainItem("TestToken")
end

function TestChest:onInspect(inspector)
    return "A simple chest used for tests.\n"
end
